File: Hour 3/Q2/mirror_sentence.c
#include<stdio.h>
#include<string.h>
#include<ctype.h>

void main(){
    char sentence[] = "Coding is awesome";
    int i, j,k=0, len, startIndex, endIndex;
    len = strlen(sentence);
    endIndex = len - 1;
    for(i = len - 1; i >= 0; i--){
        if(sentence[i] == ' ' || i == 0){
            if(i == 0)
                startIndex = 0;
            else
                startIndex = i+1;
            for(j = startIndex; j <= endIndex; j++,k++)
                printf("%c", sentence[j]);
            if(i)
                printf(" ");
            endIndex = i-1;   
        }
    }
}

File: Hour 3/Q2/mirror_sentence.java
import java.util.Scanner;

public class mirror_sentence {
    
    public static String rev_sentence(String sentence){
        String rev_sentence = "";
        String word = "";
        for(int i = 0; i < sentence.length(); i++){
            if(sentence.charAt(i) == ' '){
                rev_sentence = " " + word + rev_sentence;
                word = "";
            }
            else
                word += sentence.charAt(i);
        }
        rev_sentence = word + rev_sentence;
        return rev_sentence;
    }
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        try{
            String sentence = "Coding is awesome";
            // sentence = sc.nextLine();
            System.out.print(rev_sentence(sentence));
        }
        finally{
            sc.close();
        }
    }
}
